File: RPi Experiments/RPi2-1/C/06_potLed.c
#include<wiringPi.h>
#include<stdio.h>
#include<pcf8591.h>

#define PCF 120 //base pin for pcf8591
#define ledPin 1 //wiringPi pin number for the LED, only 1 and 24 support PWM

int main(void)
{
    int value;
    int dc;

    if(wiringPiSetup() == -1)
	{
		printf("Setup wiring Pi failed!");
		return 1;
    }
	
	pinMode(ledPin, PWM_OUTPUT); //initialize the LED pin as PWM output
    
    pcf8591Setup(PCF, 0x48); //setup pcf8591 on address 0x48

    while(1)
    {
        value = analogRead(PCF + 0); //read the value from potentiometer
        dc = value * 4; //duty cycle counts goes from 0 to 1024, the potentiometer value from 0 to 255
        printf("Value: %d\n", value);
        printf("DC: %d\n", dc);
        pwmWrite(ledPin, dc); //set the LED brightness, the duty cycle counts range goes from 0 to 1024, the potentiometer value from 0 to 255
        analogWrite(PCF + 0, value);
        delay(50); //wait for 50 ms
    }

    return 0;
}

File: RPi Experiments/RPi2-1/Python/06_potLed.py
import RPi.GPIO as GPIO
import time
import PCF8591 as ADC

ledPin = 12 # physical pin number for the LED, only 12 and 35 support PWM

def setup():
    global p
    GPIO.setwarnings(False) # don't show warnings if GPIO is in use
    GPIO.setmode(GPIO.BOARD) # use GPIO by physical location
    GPIO.setup(ledPin,GPIO.OUT) # initialize the LED pin as output
    ADC.setup(0x48)

    p = GPIO.PWM(ledPin, 100) # setup the pin for PWM, 100 Hz is enough not to see a blinking
    p.start(0) # start the PWM instance with 0 duty cycle

def loop():
    while True:
        value = ADC.read(0) # read the potentiometer value
        dc = value / 2.55 # duty cycle goes from 0 to 100, the potentiometer value from 0 to 255
        print 'Value: ' + str(value)
        print 'DC: ' + str(dc)
        p.ChangeDutyCycle(dc) # set the brightness, changing duty cycle to the value of dc 
        ADC.write(value)
        time.sleep(0.05) # wait for 50 ms

def destroy():
    p.stop() # stop the PWM instance
    ADC.write(0)
    GPIO.output(ledPin,GPIO.HIGH) # turn off led
    GPIO.cleanup() # release resource

if __name__ == '__main__': # program start from here
    try:
        setup()
        loop()

    except KeyboardInterrupt: # when 'Ctrl+C' is pressed, the child program destroy() will be  executed.
        destroy()